src/common/modal/Modal.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import ReactModal from 'react-modal';
+import classNames from 'classnames';
+
+import styles from './modal.module.scss';
+import Section from '../section/Section';
+
+interface ModalProps {
+  isOpen: boolean;
+  label: string;
+  toggleModal?: (value: boolean) => void;
+  setFormIsFilling?: (value: boolean) => void;
+  showHeading?: boolean;
+  className?: string;
+}
+
+const Modal: React.FunctionComponent<ModalProps> = ({
+  isOpen,
+  label,
+  children,
+  toggleModal,
+  setFormIsFilling,
+  className,
+}) => {
+  const onClose = () => {
+    if (setFormIsFilling) {
+      setFormIsFilling(false);
+    }
+    if (toggleModal) {
+      toggleModal(false);
+    }
+  };
+  return (
+    <>
+      {isOpen && (
```

Comment:
Are you sure we need this check here? Doesn't `isOpen={isOpen}` do the trick?

Reply:
I can be blamed for this one! I copied the logic from Kukkuu, it appears that this is indeed redundant.
https://github.com/City-of-Helsinki/kukkuu-ui/blob/2234322ac451d31e2599c01522bf914bbb8fcf35/src/common/components/modal/Modal.tsx

Reply:
Removed in this branch.